research/flamby/fed_isic2019/ditto_mkmmd/run_fold_experiment.slrm

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+
+#SBATCH --nodes=1
+#SBATCH --ntasks=1
+#SBATCH --ntasks-per-node=1
+#SBATCH --cpus-per-task=8
+#SBATCH --gres=gpu:1
+#SBATCH --mem=32G
+#SBATCH --partition=a40
+#SBATCH --qos=m
+#SBATCH --job-name=fl_five_fold_exp
+#SBATCH --output=%j_%x.out
+#SBATCH --error=%j_%x.err
+#SBATCH --time=12:00:00
+
+###############################################
+# Usage:
+#
+# sbatch research/flamby/fed_isic2019/ditto_mkmmd/run_fold_experiment.slrm \
+#   path_to_config.yaml \
+#   path_to_folder_for_artifacts/ \
+#   path_to_folder_for_dataset/ \
+#   path_to_desired_venv/ \
+#   client_side_learning_rate_value \
+#   lambda value \
+#   mu value \
+#   l2 value \
+#   mkmmd_loss_depth value \
+#   server_address
+#
+# Example:
+# sbatch research/flamby/fed_isic2019/ditto_mkmmd/run_fold_experiment.slrm \
+#   research/flamby/fed_isic2019/ditto_mkmmd/config.yaml \
+#   research/flamby/fed_isic2019/ditto_mkmmd/hp_results/ \
+#   /Users/david/Desktop/FLambyDatasets/fedisic2019/ \
+#   /h/demerson/vector_repositories/fl4health_env/ \
+#   0.0001 \
+#   0.01 \
+#   0.1 \
+#   0.0.0.0:8080
+#
+# Notes:
+# 1) The sbatch command above should be run from the top level directory of the repository.
+# 2) This example runs ditto. As such the data paths and python launch commands are hardcoded. If you want to change
+# the example you run, you need to explicitly modify the code below.
+# 3) The logging directories need to ALREADY EXIST. The script does not create them.
+###############################################
+
+# Note:
+#	  ntasks: Total number of processes to use across world
+#	  ntasks-per-node: How many processes each node should create
+
+# Set NCCL options
+# export NCCL_DEBUG=INFO
+# NCCL backend to communicate between GPU workers is not provided in vector's cluster.
+# Disable this option in slurm.
+export NCCL_IB_DISABLE=1
+
+if [[ "${SLURM_JOB_PARTITION}" == "t4v2" ]] || \
+    [[ "${SLURM_JOB_PARTITION}" == "rtx6000" ]]; then
+    echo export NCCL_SOCKET_IFNAME=bond0 on "${SLURM_JOB_PARTITION}"
+    export NCCL_SOCKET_IFNAME=bond0
+fi
+
+# Process Inputs
+
+SERVER_CONFIG_PATH=$1
+ARTIFACT_DIR=$2
+DATASET_DIR=$3
+VENV_PATH=$4
+CLIENT_LR=$5
+LAM_VALUE=$6
+MU_VALUE=$7
+L2_VALUE=$8
+MKMMD_LOSS_DEPTH=$9
+BETA_INTERVAL=${10}
+SERVER_ADDRESS=${11}
```

Comment:
Very minor, as I don't think this actually has an effect, but is there a reason to use ${.} for the last two arguments?

Reply:
If I don't put {} it will understand it as $1+1.  For example if $1 is "Book", the argument would be  "Book1".

Reply:
Man bash is dumb sometimes 😂. Fair enough!